types/definition/definition.go

```diff
@@ -7,11 +7,11 @@ import (
 )
 
 func IsMapType(fieldType string) bool {
-	return strings.HasPrefix(fieldType, "map[") && strings.HasSuffix(fieldType, "]")
+	return (strings.HasPrefix(fieldType, "map[") || strings.HasPrefix(fieldType, "nullablemap[")) && strings.HasSuffix(fieldType, "]")
 }
 
 func IsArrayType(fieldType string) bool {
-	return strings.HasPrefix(fieldType, "array[") && strings.HasSuffix(fieldType, "]")
+	return (strings.HasPrefix(fieldType, "array[") || strings.HasPrefix(fieldType, "nullablearray[")) && strings.HasSuffix(fieldType, "]")
```

Comment:
I guess it should, for the sake of completeness. I'm not 100% sure what the implications would be for generalizing it to the `reference[]` type. It doesn't exactly map to a regular Go type like slices and maps, I can't picture a case where it would make sense to use a `nullablereference[]`. But it has to be opted into so it shouldn't hurt anything.

Reply:
Maybe if a field is optional but otherwise should reference an existing type? If we don't intend for it to be used that way let's just remove it; removing it should cause an error to be returned when the type can't be parsed. We can always add the functionality if we end up wanting it. My bad.

Reply:
Removed it